pkg/registry/directoryGraphLoader.go

```diff
@@ -0,0 +1,225 @@
+			if err == nil {
+				replaceStrings = append(replaceStrings, r)
+			}
+
+			g.CsvNameAndReplaceMap[csvStruct.GetName()] = csvReplaces{
+				replaces:  replaceStrings,
+				skipRange: sr,
+			}
+
+			version, err := csvStruct.GetVersion()
+			if err == nil && version != "" {
+				v, err := semver.Parse(version)
+				if err == nil {
+					g.SortedCSVs = append(g.SortedCSVs, csv{name: csvStruct.GetName(), version: v})
+				}
+			}
+		}
+	}
+
+	sort.Sort(g.SortedCSVs)
+	return nil
+}
+
+// loadReplaces converts skipRange to explicit csv name and write to replaces in CsvNameAndReplaceMap.
+func (g *DirGraphLoader) loadReplaces() {
+	for csvName, replace := range g.CsvNameAndReplaceMap {
+		if replace.skipRange != nil {
+			for _, v := range g.SortedCSVs {
+				if replace.skipRange(v.version) {
+					g.CsvNameAndReplaceMap[csvName] = csvReplaces{
+						replaces:  append(g.CsvNameAndReplaceMap[csvName].replaces, v.name),
+						skipRange: g.CsvNameAndReplaceMap[csvName].skipRange,
+					}
+				}
+			}
+		}
+	}
+}
+
+// getChannelNodes follows the head of the channel csv through all replaces to fill the nodes until the tail of the
+// channel which will not replace anything.
+func (g *DirGraphLoader) getChannelNodes(channelHeadCsv string) *map[BundleKey]map[BundleKey]struct{} {
+	nodes := make(map[BundleKey]map[BundleKey]struct{})
+	remainingCSVsInChannel := make(map[BundleKey]struct{})
+
+	for _, csv := range g.CsvNameAndReplaceMap[channelHeadCsv].replaces {
+		remainingCSVsInChannel[BundleKey{CsvName: csv}] = struct{}{}
+	}
+
+	// Iterate through remainingCSVsInChannel and add replaces of each encountered CSVs if not already in nodes.
+	// Loop only exit after all remaining csvs are visited/deleted.
+	for len(remainingCSVsInChannel) > 0 {
+		for bk, _ := range remainingCSVsInChannel {
+			if _, ok := nodes[BundleKey{CsvName: channelHeadCsv}]; !ok {
+				nodes[BundleKey{CsvName: channelHeadCsv}] = func() map[BundleKey]struct{} {
+					subNode := make(map[BundleKey]struct{})
+					for _, csv := range g.CsvNameAndReplaceMap[channelHeadCsv].replaces {
+						subNode[BundleKey{CsvName: csv}] = struct{}{}
+					}
+					return subNode
+				}()
+				for _, csv := range g.CsvNameAndReplaceMap[channelHeadCsv].replaces {
+					if _, ok := nodes[BundleKey{CsvName: csv}]; !ok {
+						remainingCSVsInChannel[BundleKey{CsvName: csv}] = struct{}{}
+					}
+				}
+			}
+			delete(remainingCSVsInChannel, bk)
+		}
+	}
+	return &nodes
+}
+
+// parsePackageYAMLFile parses the *.package.yaml file and fills the information in Package including name,
+// defaultchannel, and head of all Channels. It returns parsing error if any.
+func (g *DirGraphLoader) parsePackageYAMLFile() (*Package, error) {
+	files, err := ioutil.ReadDir(g.PackageDir)
+	if err != nil {
+		return nil, fmt.Errorf("error reading bundle parent directory, %v", err)
+	}
+
+	ymlFiles := []string{}
+	for _, f := range files {
+		if matched, err := path.Match("*.yaml", f.Name()); err != nil {
```

Comment:
can we just check to see if it's yaml? we do something similar when we unpack bundles: https://github.com/operator-framework/operator-registry/blob/master/pkg/registry/conversion.go#L67